src/ConstructiveSolidGeometry/SurfacePrimitives/EllipticalSurface.jl

```diff
@@ -0,0 +1,178 @@
+"""
+    EllipticalSurface{T,TR,TP} <: AbstractSurfacePrimitive{T}
+
+* `r::TR`: 
+    * TR = Real -> Full Circle (a = b = r)
+    * TR = (Real, Real) -> Circular Annulus (r_in = r[1], r_out = r[2])
+    * TR = ((Real,), (Real,)) -> Full Ellipse (a = r[1][1], b = r[2][1])
+    * TR = ((Real, Real),(Real, Real)) -> Elliptical Annulus \n(a_in = r[1][1], a_out = r[1][2], b_in = r[2][1], b_out = r[2][2])
+    * Not all are implemented yet
+
+* `φ::TP`: 
+    * TP = Nothing <-> Full in φ
+    * ...
+"""
+@with_kw struct EllipticalSurface{T,TR,TP} <: AbstractPlanarSurfacePrimitive{T}
+    r::TR = 1
+    φ::TP = nothing
+
+    origin::CartesianPoint{T} = zero(CartesianPoint{T})
+    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+end
+
+const CircularArea{T} = EllipticalSurface{T,T,Nothing}
+const PartialCircularArea{T} = EllipticalSurface{T,T,Tuple{T,T}}
+
+const Annulus{T} = EllipticalSurface{T,Tuple{T,T},Nothing}
+const PartialAnnulus{T} = EllipticalSurface{T,Tuple{T,T},Tuple{T,T}}
+
+Plane(es::EllipticalSurface{T}) where {T} = Plane{T}(es.origin, es.rotation * CartesianVector{T}(zero(T),zero(T),one(T)))
+
+normal(es::EllipticalSurface{T}) where {T} = es.rotation * CartesianVector{T}(zero(T), zero(T), one(T))
+
+extremum(es::EllipticalSurface{T,T}) where {T} = es.r
+extremum(es::EllipticalSurface{T,Tuple{T,T}}) where {T} = es.r[2] # r_out always larger r_in: es.r[2] > es.r[2]
```

Comment:
Why was that never used in the primitives above?

Reply:
Do you mean why not `max(es.r...)`?
Here, it it contains `r_in` and `r_out`. 
For the above primitives it was, e.g., `r_bot` and `r_top`.